recipes/circularbuffer/all/test_package/conanfile.py

```diff
@@ -1,16 +1,26 @@
+from conan import ConanFile
+from conan.tools.build import cross_building
+from conan.tools.cmake import CMake, cmake_layout
+
 import os
-from conans import ConanFile, CMake, tools
 
-class TestConan(ConanFile):
+
+class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "cmake_find_package", "cmake_find_package_multi"
```

Comment:
`test_package`(v2) should use CMakeToolchain, CMakeDeps and VirtualRunEnv https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/header_only/all/test_package/conanfile.py#L10
```suggestion
    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
    test_type = "explicit"
```